### PR TITLE
[Ninja] fix mingw parse format

### PR DESCRIPTION
**Pull request by dmoody256**
This was observed in the mail chain https://pairlist4.pair.net/pipermail/scons-users/2022-April/008903.html

## Contributor Checklist:

* [x] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [x] I have updated `CHANGES.txt` (and read the `README.rst`)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -49,6 +49,8 @@ RELEASE  VERSION/DATE TO BE FILLED IN LATER
     - Ninja: Fix issue where Configure files weren't being properly processed when build run
       via ninja.
     - Added ninja mingw support and improved ninja CommandGeneratorAction support.
+    - Added user configurable setting of ninja depfile format. This helps in mingw environments
```
Comment (bdbaddog) [inline CHANGES.txt]: please add `(via NINJA_DEPFILE_PARSE_FORMAT)` to this and the release.txt blurb

**Comment by bdbaddog** [comment]
Also please create a test and then should be good to merge.

**Comment by dmoody256** [comment]
Okay test added in the latest

File: CHANGES.txt (hunk 1 above)
Comment (dmoody256) [reply CHANGES.txt]: updated in the latest

**Comment by dmoody256** [comment]
the test specifically checks for gcc format, so if you want it run on all platforms then it should make that check conditional on the platform

**Pull request merged**